charmcraft/application/main.py

```diff
@@ -127,14 +127,30 @@ def _get_dispatcher(self) -> craft_cli.Dispatcher:
     def _get_app_plugins(self) -> dict[str, PluginType]:
         return parts.get_app_plugins()
 
+    @override
```

Comment:
This is a workaround until we can upgrade to craft-application 5.6.4: https://github.com/canonical/craft-application/pull/857